Project-Aurora/Project-Aurora/Devices/Uniwill/KeyboardDevice.cs

```diff
@@ -0,0 +1,373 @@
+        }
+   
+
+        private void OnRegChanged(object sender, EventArgs e)
+        {
+            
+            if (CheckGCPower())
+            {
+                Initialize();
+            } 
+            else
+            {
+                bRefreshOnce = true;
+                isInitialized = false;
+                Shutdown();
+            }
+            
+        }
+
+        public string GetDeviceName()
+            {
+                return devicename;
+            }
+
+            public string GetDeviceDetails()
+            {
+                if (isInitialized)
+                {
+                    return devicename + ": Initialized";
+                }
+                else
+                {
+                    return devicename + ": Not initialized";
+                }
+            }
+
+            public bool Initialize()
+            {
+           
+
+              if (!isInitialized && CheckGCPower())
+                {
+                    try
+                    {
+                     
+                        keyboard =   KeyboardFactory.CreateHIDDevice("hidkeyboard");
+                   
+                    if (keyboard!=null)
+                        {
+                            isInitialized = true;
+                            return true;
+                        }
+                           
+                    
+                    isInitialized = false;
+                    return false;
+                    // Mark Initialized = TRUE
+                   
+                    }
+                    catch (Exception ex)
+                    {
+                        Global.logger.Error("keyboard device, Exception! Message:" + ex);
+                    }
+                    //try
+                    //{
+                    //    // Initialize  
+                       
+
+
+                    //    // Mark Initialized = TRUE
+                    //    isInitialized = true;
+                    //    return true;
+                    //}
+                    //catch (Exception ex)
+                    //{
+                    //    Global.logger.Error("keyboard device, Exception! Message:" + ex);
+                    //}
+
+
+
+                   // Mark Initialized = FALSE
+                   isInitialized = false;
+                    return false;
+                }
+ 
+
+            return isInitialized;
+
+            }
+
+            // Handle Logon Event
+            //void SystemEvents_SessionSwitch(object sender, SessionSwitchEventArgs e)
+            //{
+            //    if (this.IsInitialized() && e.Reason.Equals(SessionSwitchReason.SessionUnlock))
+            //    { // Only Update when Logged In
+            //        this.SendColorsToKeyboard(true);
+            //    }
+            //}
+
+            public void Shutdown()
+            {
+                if (this.IsInitialized())
+                {
+                    if(CheckGCPower())
+                     keyboard?.release();
```

Comment:
Is this supposed to be the same code as Reset()? If yes why not just call Reset() here?

Reply:
I believe this is checking their own software.